File: lua/ht/plugins/ui/nvim-scrollview.lua
return {
  -- scroll
  {
    "dstein64/nvim-scrollview",
    lazy = true,
    enabled = false,
    event = { "BufReadPost" },
    dependencies = {
      "kevinhwang91/nvim-hlslens",
      "folke/tokyonight.nvim",
    },
    opts = {},
    config = function()
      require("scrollview").setup {
        excluded_filetypes = {
          "prompt",
          "TelescopePrompt",
          "noice",
          "toggleterm",
          "nuipopup",
          "NvimTree",
          "rightclickpopup",
        },
        current_only = true,
        winblend = 75,
        base = "right",
        column = 2,
        signs_on_startup = { "all" },
        diagnostics_severities = { vim.diagnostic.severity.WARN },
      }
    end,
  },
}

File: src/conf/plugins/ui/nvim-scrollview.ts
import { Plugin, PluginOptsBase } from "@core/model";

const spec: PluginOptsBase = {
  shortUrl: "dstein64/nvim-scrollview",
  lazy: {
    enabled: false,
    event: ["BufReadPost"],
    dependencies: ["kevinhwang91/nvim-hlslens", "folke/tokyonight.nvim"],
    opts: {
      excluded_filetypes: [
        "prompt",
        "TelescopePrompt",
        "noice",
        "toggleterm",
        "nuipopup",
        "NvimTree",
        "rightclickpopup",
      ],
      current_only: true,
      winblend: 75,
      base: "right",
      column: 2,
      signs_on_startup: ["all"],
    },
    config: true,
  },
};

export const plugin = new Plugin(spec);
